core-blocks/file/edit.js

```diff
@@ -0,0 +1,252 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { getBlobByURL, revokeBlobURL } from '@wordpress/utils';
+import {
+	ClipboardButton,
+	IconButton,
+	Toolbar,
+	withNotices,
+} from '@wordpress/components';
+import { Component, compose, Fragment } from '@wordpress/element';
+import {
+	MediaUpload,
+	MediaPlaceholder,
+	BlockControls,
+	RichText,
+	editorMediaUpload,
+} from '@wordpress/editor';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+import FileBlockInspector from './inspector';
+import FileBlockEditableLink from './editable-link';
+
+class FileEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		const {
+			href,
+			showDownloadButton = true,
+			buttonText = __( 'Download' ),
+			id,
+		} = this.props.attributes;
+
+		this.onSelectFile = this.onSelectFile.bind( this );
+
+		// Initialize default values if undefined
+		this.props.setAttributes( {
+			showDownloadButton,
+			buttonText,
+		} );
+
+		// edit component has its own attributes in the state so it can be edited
+		// without setting the actual values outside of the edit UI
+		this.state = {
+			showPlaceholder: ! href,
+			href,
+			attachmentPage: undefined,
+			showCopyConfirmation: false,
+		};
+
+		if ( id !== undefined ) {
+			// Get Attachment Page URL
+			wp.apiRequest( {
```

Comment:
Any reason we’re not using [withAPIData](https://github.com/WordPress/gutenberg/tree/master/components/higher-order/with-api-data) here?

Reply:
I didn't realize we had that! Cool. I put that in place, however it turns out that the snapshot test helper for our components (`blockEditRender()`) isn't equipped to inject `context` in order for it to work. I've made a patch for it and I'll propose it in a separate PR, but not sure if my implementation is good. Anyway, I'm going to remove the snapshot test for this temporarily.